src/predicates.cpp

```diff
@@ -117,4 +117,77 @@ void init_predicates(py::module& m) {
             Geography object(s)
 
     )pbdoc");
+
+    m.def("touches",
+          py::vectorize(Predicate([](const s2geog::ShapeIndexGeography& a_index,
+                                     const s2geog::ShapeIndexGeography& b_index,
+                                     const S2BooleanOperation::Options& options) {
+              S2BooleanOperation::Options closedOptions = options;
+              closedOptions.set_polygon_model(S2BooleanOperation::PolygonModel::CLOSED);
+              closedOptions.set_polyline_model(S2BooleanOperation::PolylineModel::CLOSED);
+
+              S2BooleanOperation::Options openOptions = options;
+              openOptions.set_polygon_model(S2BooleanOperation::PolygonModel::OPEN);
+              openOptions.set_polyline_model(S2BooleanOperation::PolylineModel::OPEN);
```

Comment:
Agreed. However, especially in this case, do you think exposing `m_options` would be sufficient? I think we would actually need two `S2BooleanOperation::Options` objects with different 'settings' here to properly do the check.

Would it make sense to maybe move away from having the `Predicate` class and have something like one specialized struct per predicate? Considering also that there may be a few more edge cases and having a bit more flexibility in the implementation might make it easier to not affect other predicates?

Reply:
Ah yes, for this case having a separate class will work better (didn't think about the touches case doing two calls with different options). For the others it might work fine with the current Predicate structure.

But certainly feel free to add a separate class for TouchesPredicate and we can see from there

Reply:
I have done so and exposed the options on the Predicate functor. The initialization should now happen only once.